rental.rb

```diff
@@ -0,0 +1,22 @@
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
+
+  def initialize(date, person, book)
+    @date = date
+    @person = person
+    @book = book
+    book.rentals_list.push(self)
+    person.rentals_list.push(self)
+  end
+
+  def add_person(person)
+    @person = person
+    person.rentals_list.push(self) unless person.rentals_list.include?(self)
+  end
```

Comment:
- [ ] There is no need for the add_person method in the Rental class since you defined the instance variable `@person` and pushed the rental itself to the person rentals_list in the initialize method:
```rb
  def initialize(date, person, book)
    @date = date
    @person = person #here
    @book = book
    book.rentals_list.push(self)  
    person.rentals_list.push(self) # and here
  end
```